chew.go

```diff
@@ -148,6 +349,10 @@ func Process(urls []string, ctxs ...context.Context) ([]common.Chunk, error) {
 	return result, nil
 }
 
+/*
+processURL handles the actual processing of a single URL or file
+file paths are processed directly while URLs are fetched and processed
+*/
 func processURL(url string, ctxs ...context.Context) ([]common.Chunk, error) {
```

Comment:
it's an internal function, you should refactor it

ctxs as a variadic is not a good thing

You should have this

```suggestion
func processURL(ctx context.Context, url string) ([]common.Chunk, error) {
```

Reply:
I didn't actually know that this was the case, initially I made it like this so that the context itself would be optional